datafusion/physical-plan/src/aggregates/mod.rs

```diff
@@ -3018,4 +3019,112 @@ mod tests {
         run_test_with_spill_pool_if_necessary(20_000, false).await?;
         Ok(())
     }
+
+    // Fix for https://github.com/apache/datafusion/issues/15530
+    #[tokio::test]
+    async fn test_single_mode_aggregate_with_spill() -> Result<()> {
```

Comment:
Since this is a fuzz test, maybe we can put it with the rest of the fuzz code: https://github.com/apache/datafusion/blob/main/datafusion/core/tests/fuzz_cases/aggregate_fuzz.rs

Reply:
done